pandas/core/groupby/generic.py

```diff
@@ -1253,34 +1253,7 @@ def first_not_none(values):
 
             if isinstance(v, (np.ndarray, Index, Series)):
                 if isinstance(v, Series):
-                    applied_index = self._selected_obj._get_axis(self.axis)
```

Comment:
For deprecations, we can only remove this once we release pandas 2.0. `squeeze` must continue to work up until then, so we cannot remove all this code just yet.

Reply:
Oh sorry, I did not know that.

Just to get that right:
So I can only add a warning in a Pull Request, which will be merged into the master?

This Pull Request could be added before releasing 2.0 but not earlier?

Reply:
> So I can only add a future warning in a Pull Request, which will be merged into the master?

Correct.

> This Pull Request could be merged into master before releasing 2.0 but not earlier?

This pull request with just the futurewarnings can be merged into master before releasing 2.0

So another pull request will need to be added in the future where this code is removed when we have decided that 2.0 will be the next release.

Reply:
Thanks very much for the clarification. I will modify the PR accordingly tomorrow.

Reply:
I reverted the actual code changes and kept the warning.